src/Deserializers/EntityDeserializer.php

```diff
@@ -59,6 +59,11 @@ public function deserialize( $serialization ) {
 	 */
 	protected abstract function getPartiallyDeserialized( array $serialization );
 
+	/**
+	 * @param array $serialization
+	 *
+	 * @return Entity
```

Comment:
I tried but it's not that easy. The whole class depends on Entity. Please merge this and try to phase out Entity in an other patch.